### ISSUE TITLE
Qt issue when using default GR_jll

### ISSUE THREAD

**Issue by luraess** [jheinen/GR.jl#454]
Using Plots.jl with the GR backend (default) fails to open a GKS Qt Term plotting window on Ubuntu 20.04, with Julia 1.7.2, and Plots v1.27.6 (having latest Qt5-base and Ubuntu 20.04 up to date).
```
(@v1.7) pkg> add Plots

julia> using Plots

julia> heatmap(rand(4,3))
/home/luraess/.julia/artifacts/0f27f5d53948f64dabadd58b4f20b8a60dd806aa/bin/gksqt: /lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.15' not found (required by /home/luraess/.julia/artifacts/0f27f5d53948f64dabadd58b4f20b8a60dd806aa/bin/gksqt)
connect: Connection refused
GKS: can't connect to GKS socket application

GKS: Open failed in routine OPEN_WS
```

This seems to be the recurring issue ([since Spring 2021](https://discourse.julialang.org/t/why-does-plots-require-firewall-access-gksqt/59803/10?u=luraess)) when using the default `GR.gr_provider` pointing to `"BinaryBuilder"`. Current workaround is once more to perform following 
```
julia> ENV["JULIA_GR_PROVIDER"] = "GR"
"GR"

julia> import Pkg; Pkg.build("GR")
    Building GR -> `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/af237c08bda486b74318c8070adb96efa6952530/build.log`
Precompiling project...
  2 dependencies successfully precompiled in 36 seconds (142 already precompiled)

julia> using GR

julia> GR.gr_provider
Base.RefValue{String}("GR")

julia> using Plots
```
My understanding is that there is a (recurring?) issue with default behaviour of using the GR_jll and not the GR tarball since a year. 

Would it be possible to fix this behaviour as the above workaround is not really viable since any change in project packages would potentially re-trigger build and thus switch back to default behaviour.

Note that on macOS, the default GR_jll usage makes the GKS Qt Term window to display in a degraded mode which makes e.g. impossible to close the plotting window without closing the running Julia instance.

**Comment by jheinen** [comment]
I cannot re-produce the behaviour, neither with Ubuntu 20.04 nor with Ubuntu 22.04 (or macOS 12.3.1). I used a "fresh" Julia environment (with `BinaryBuilder` as the default GR provider) - may be I misunderstood your concern. Does the problem occur after a `Plots` update?

> Note that on macOS, the default GR_jll usage makes the GKS Qt Term window to display in a degraded mode which makes e.g. impossible to close the plotting window without closing the running Julia instance.

What do you mean with "degraded mode"?

**Comment by luraess** [comment]
Thanks for looking into it @jheinen. Weird it does not fail. Re-doing the operation in a yet other "fresh" Julia project, it indeed does not fail but Plots.jl builds against the GR tarball and not the jll.
```
(tmp_plots) pkg> st
      Status `/scratch-1/luraess/tmp_plots/Project.toml` (empty project)

(tmp_plots) pkg> add Plots
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `/scratch-1/luraess/tmp_plots/Project.toml`
  [91a5bcdd] + Plots v1.27.6
  [...]

julia> using Plots
[ Info: Precompiling Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]

julia> heatmap(rand(4,3))

julia> GR.gr_provider
Base.RefValue{String}("GR")

julia> 

```

> What do you mean with "degraded mode"?

Upon plotting, not the regular GKS Qt Term window opens but a generic window that e.g. cannot be closed without closing or crashing Julia.

**Comment by luraess** [comment]
Exactly. I am get this. For sure I could put `ENV["JULIA_GR_PROVIDER"] = "GR"` in a `startup.jl` file or define it in a shell env. But that's one more step to take track and care of, that will potentially fail for another user, etc...

A question is why the default `jll` is failing and why to have it as default if the fix is to switch to the non-default?

**Comment by jheinen** [comment]
Still don't understand why the default provider in your environment is "GR" and why the `GKSTerm.app` (on macOS) crashes. Which GR version are you using? Should be:
```julia
using GR
version()
"0.64.2"
```
Would like to help, but I can't re-produce this on any of our systems ...

**Comment by luraess** [comment]
> Still don't understand why the default provider in your environment is "GR"

It used to be "BinaryBuilder", then I changed to "GR" because of the fail and for a reason I don't know it seems to have persisted. Maybe it has to do the the fail occurred having Plots.jl in the v1.7 environment and now I was trying in a specific project.

On macOS (still on `10.15.7 Catalina`), I don't know what brings the `GKSTerm.app` not to work properly when `jll` are used.

Thanks for helping out, and I understand it may be difficult since the failure pattern is (at least for me) not easy to get in order to make it reproducible.

**Comment by jheinen** [comment]
> On macOS (still on 10.15.7 Catalina), I don't know what brings the GKSTerm.app not to work properly when jll are used.

This is even more amazing because 10.15.7 is exactly the version of our CI build system for Intel based macOS systems ...
